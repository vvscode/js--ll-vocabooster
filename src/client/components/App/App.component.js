import React from 'react';
import PropTypes from 'prop-types';

const ContextType = {};

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return Object.assign({}, this.props.context);
  }

  render() {
    return React.Children.only(this.props.children);
  }
}

export default App;
