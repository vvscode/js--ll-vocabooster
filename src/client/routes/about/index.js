import React from 'react';
import About from './About.component';
import './About.css';

const title = 'About';

export default () => ({
  title,
  component: props => <About title={title} {...props} />,
});
/*
File was generated with tamm-cli from next settings:
{
  "path": "about",
  "addStyles": true,
  "connectedToRedux": "No",
  "componentName": "about",
  "camelizedComponentName": "About",
  "withRedux": false
}
*/
