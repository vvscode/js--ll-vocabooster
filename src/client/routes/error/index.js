import React from 'react';
import ErrorPage from './Error.component';

import './Error.css';

export default () => ({
  title: 'Error',
  component: () => <ErrorPage />,
});
