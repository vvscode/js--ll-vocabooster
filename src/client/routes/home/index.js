import React from 'react';
import Home from './Home.container';

import './Home.css';

const title = 'Home';

export default () => ({
  title,
  component: props => <Home title={title} {...props} />,
});
