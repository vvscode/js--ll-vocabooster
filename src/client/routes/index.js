/* eslint-disable global-require */
import React from 'react';
import { Redirect } from 'react-router';
import { AsyncPageLoader } from 'client/components/RouteComponents';

const routes = [
  {
    path: '/',
    exact: true,
    component: AsyncPageLoader('home'),
  },
  {
    path: '/home',
    component: AsyncPageLoader('home'),
    routes: [
      {
        path: '/home/',
        exact: true,
        component: AsyncPageLoader('home/root'),
      },
      {
        path: '/home/personal',
        exact: true,
        component: AsyncPageLoader('home/root'),
      },
      {
        path: '/home/tab1',
        key: '/home/tab1',
        component: () => (
          <div>
            <h4>Tab 1</h4>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
              nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi.
              Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum.
              Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris
              massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti
              sociosqu ad litora torquent per conubia nostra, per inceptos
              himenaeos. Curabitur sodales ligula in libero.
              <i>Lorem ipsum dolor sit amet, consectetur adipiscing elit</i>.
              Sed dignissim lacinia nunc.
            </p>

            <p>
              <b>Praesent mauris</b>. Curabitur tortor. Pellentesque nibh.
              <b>Fusce nec tellus sed augue semper porta</b>. Aenean quam. In
              scelerisque sem at dolor. Maecenas mattis. Sed convallis tristique
              sem. <b>Vestibulum lacinia arcu eget nulla</b>. Proin ut ligula
              vel nunc egestas porttitor. Morbi lectus risus, iaculis vel,
              suscipit quis, luctus non, massa. Fusce ac turpis quis ligula
              lacinia aliquet. Mauris ipsum. Nulla metus metus, ullamcorper vel,
              tincidunt sed, euismod in, nibh. Quisque volutpat condimentum
              velit. Class aptent taciti sociosqu ad litora torquent per conubia
              nostra, per inceptos himenaeos.
            </p>

            <p>
              Nam nec ante. Sed lacinia, urna non tincidunt mattis, tortor neque
              adipiscing diam, a cursus ipsum ante quis turpis. Nulla facilisi.
              Ut fringilla. Suspendisse potenti. Nunc feugiat mi a tellus
              consequat imperdiet. Vestibulum sapien. Proin quam. Etiam
              ultrices. Suspendisse in justo eu magna luctus suscipit. Sed
              lectus. Integer euismod lacus luctus magna.
            </p>

            <p>
              Quisque cursus, metus vitae pharetra auctor, sem massa mattis sem,
              at interdum magna augue eget diam. Vestibulum ante ipsum primis in
              faucibus orci luctus et ultrices posuere cubilia Curae; Morbi
              lacinia molestie dui. Praesent blandit dolor. Sed non quam. In vel
              mi sit amet augue congue elementum. Morbi in ipsum sit amet pede
              facilisis laoreet. Donec lacus nunc, viverra nec, blandit vel,
              egestas et, augue. Vestibulum tincidunt malesuada tellus. Ut
              ultrices ultrices enim. Curabitur sit amet mauris. Morbi in dui
              quis est pulvinar ullamcorper. Nulla facilisi. Integer lacinia
              sollicitudin massa.
            </p>

            <p>
              Cras metus. Sed aliquet risus a tortor. Integer id quam. Morbi mi.
              Quisque nisl felis, venenatis tristique, dignissim in, ultrices
              sit amet, augue. Proin sodales libero eget ante. Nulla quam.
              Aenean laoreet. Vestibulum nisi lectus, commodo ac, facilisis ac,
              ultricies eu, pede. Ut orci risus, accumsan porttitor, cursus
              quis, aliquet eget, justo. Sed pretium blandit orci. Ut eu diam at
              pede suscipit sodales. Aenean lectus elit, fermentum non,
              convallis id, sagittis at, neque. Nullam mauris orci, aliquet et,
              iaculis et, viverra vitae, ligula.
            </p>
          </div>
        ),
      },
      {
        path: '/home/tab2',
        key: '/home/tab2',
        component: () => (
          <div>
            <h4>Tab 2</h4>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
              nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi.
              Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum.
              <b>Lorem ipsum dolor sit amet, consectetur adipiscing elit</b>.
              Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris
              massa. Vestibulum lacinia arcu eget nulla.
              <b>Lorem ipsum dolor sit amet, consectetur adipiscing elit</b>.
              Class aptent taciti sociosqu ad litora torquent per conubia
              nostra, per inceptos himenaeos. Curabitur sodales ligula in
              libero.
              <b>Lorem ipsum dolor sit amet, consectetur adipiscing elit</b>.
              Sed dignissim lacinia nunc. Curabitur tortor. Pellentesque nibh.
              Aenean quam. In scelerisque sem at dolor. Maecenas mattis. Sed
              convallis tristique sem. Proin ut ligula vel nunc egestas
              porttitor. Morbi lectus risus, iaculis vel, suscipit quis, luctus
              non, massa.
              <b>
                Class aptent taciti sociosqu ad litora torquent per conubia
                nostra, per inceptos himenaeos
              </b>. Fusce ac turpis quis ligula lacinia aliquet. Mauris ipsum.
              Nulla metus metus, ullamcorper vel, tincidunt sed, euismod in,
              nibh. Quisque volutpat condimentum velit. Class aptent taciti
              sociosqu ad litora torquent per conubia nostra, per inceptos
              himenaeos. Nam nec ante. Sed lacinia, urna non tincidunt mattis,
              tortor neque adipiscing diam, a cursus ipsum ante quis turpis.
              Nulla facilisi. <b>Proin ut ligula vel nunc egestas porttitor</b>.
              Ut fringilla. Suspendisse potenti. Nunc feugiat mi a tellus
              consequat imperdiet. Vestibulum sapien. Proin quam. Etiam
              ultrices. <b>Nam nec ante</b>. Suspendisse in justo eu magna
              luctus suscipit. Sed lectus. Integer euismod lacus luctus magna.
              Quisque cursus, metus vitae pharetra auctor, sem massa mattis sem,
              at interdum magna augue eget diam. Vestibulum ante ipsum primis in
              faucibus orci luctus et ultrices posuere cubilia Curae; Morbi
              lacinia molestie dui. Praesent blandit dolor. Sed non quam. In vel
              mi sit amet augue congue elementum. Morbi in ipsum sit amet pede
              facilisis laoreet. Donec lacus nunc, viverra nec, blandit vel,
              egestas et, augue. Vestibulum tincidunt malesuada tellus. Ut
              ultrices ultrices enim. Curabitur sit amet mauris. Morbi in dui
              quis est pulvinar ullamcorper.
            </p>
          </div>
        ),
      },
      {
        component: () => <Redirect to="/404" />,
      },
    ],
  },
  {
    component: AsyncPageLoader('not-found'),
  },
];

// The error page is available by permanent url for development mode
if (__DEV__) {
  routes.push({
    path: '/error',
    component: AsyncPageLoader('error'),
  });
}

export default routes;
