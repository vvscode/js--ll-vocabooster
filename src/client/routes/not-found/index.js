import React from 'react';
import NotFound from './NotFound.component';

import './NotFound.css';

const title = 'Error - 404';

export default () => ({
  title,
  component: props => <NotFound title={title} {...props} />,
});
