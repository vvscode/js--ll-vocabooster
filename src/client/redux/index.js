import { applyMiddleware, combineReducers, createStore } from 'redux';
import { createLogger as reduxLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import thunk from 'redux-thunk';
import Api from 'client/services/InternalApi';
import ErrorHandler from 'client/services/ErrorHandler';

import layout from './layout';
import home from './home';

export const rootReducer = combineReducers({
  [home.name]: home.reducer,
  [layout.name]: layout.reducer,
});

const middlewares = [thunk.withExtraArgument({ Api, ErrorHandler })];

export const configureStore = () => {
  const composeEnhancers = composeWithDevTools({
    // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
    // https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md
  });

  if (__DEV__) {
    // https://github.com/evgenyrodionov/redux-logger#options
    middlewares.push(reduxLogger({ collapsed: true }));
  }

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middlewares)),
  );
};
