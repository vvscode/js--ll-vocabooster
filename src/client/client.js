import 'whatwg-fetch';
import React from 'react';
import ReactDOM from 'react-dom';
import deepForceUpdate from 'react-deep-force-update';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import { Router, Switch } from 'react-router';

import history from 'client/utils/history';
import Application from 'client/components/App';
import Layout from 'client/containers/Layout/';
import routes from 'client/routes';
import { configureStore } from 'client/redux';

// Global (context) variables that can be easily accessed from any React component
// https://facebook.github.io/react/docs/context.html
const context = {};

const container = document.getElementById('app');
let appInstance;

const store = configureStore();

async function onLocationChange() {
  try {
    // TODO: search alternative solution
    const renderReactApp = ReactDOM.render;
    appInstance = renderReactApp(
      <Application context={context}>
        <Provider store={store}>
          <Router history={history}>
            <Layout>
              <Switch>{renderRoutes(routes)}</Switch>
            </Layout>
          </Router>
        </Provider>
      </Application>,
      container,
    );
  } catch (error) {
    if (__DEV__) {
      throw error;
    }
    console.error(error);
  }
}
onLocationChange();

// Enable Hot Module Replacement (HMR)
if (module.hot) {
  module.hot.accept('./routes', () => {
    if (appInstance && appInstance.updater.isMounted(appInstance)) {
      // Force-update the whole tree, including components that refuse to update
      deepForceUpdate(appInstance);
    }

    onLocationChange();
  });
}
