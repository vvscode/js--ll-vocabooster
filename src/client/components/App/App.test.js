/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { shallow } from 'enzyme';
import 'client/routes';
import App from './App.component';

describe('App', () => {
  test('renders children correctly', () => {
    const context = {};
    const wrapper = shallow(
      <App context={context}>
        <div className="child" />
      </App>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
