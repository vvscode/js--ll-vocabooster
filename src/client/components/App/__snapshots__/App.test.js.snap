// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders children correctly 1`] = `
<div
  className="child"
/>
`;
